### PR TITLE
Add docstrings in mlflow logger class

### LINKED ISSUE #9028: add missing docstrings to public properties in loggers
## 📚 Documentation

There are a few undocumented public properties in the logger wrappers. 
Example: 
https://github.com/PyTorchLightning/pytorch-lightning/blob/92e49795e1e044ee7f1ca450babf0c50c1b5d81f/pytorch_lightning/loggers/mlflow.py#L174

Let's document these so they show up in our HTML docs and so users can discover them. 

**This is a good issue for new contributors!! If you are interested, please send a pull request right away (one per logger class).**

### PR DESCRIPTION
**Pull request by bamblebam**
## What does this PR do?

Fixes #9028 
Adds docstrings in the mlfow logger class.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/mlflow.py [hunk 1]
```diff
@@ -172,13 +172,19 @@ def experiment(self) -> MlflowClient:
 
     @property
     def run_id(self):
-        # create the experiment if it does not exist to get the run id
+        """
+        Create the experiment if it does not exist to get the run id.
+        Return: The run id.
```
Comment (awaelchli) [inline pytorch_lightning/loggers/mlflow.py:177]:
```suggestion
        Create the experiment if it does not exist to get the run id.
        
        Return:
            The run id.
```
for the google docstring style we need spacing here so it gets rendered properly :)

File: pytorch_lightning/loggers/mlflow.py [hunk 2]
```diff
@@ -172,13 +172,19 @@ def experiment(self) -> MlflowClient:
...
+        """
         _ = self.experiment
         return self._run_id
 
     @property
     def experiment_id(self):
```
Comment (awaelchli) [inline pytorch_lightning/loggers/mlflow.py:183]: can we also add the type here while we are at it?

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/loggers/mlflow.py (hunk 1 above)
Comment (bamblebam) [reply pytorch_lightning/loggers/mlflow.py]: I did this and also added the return types.

Comment (awaelchli) [reply pytorch_lightning/loggers/mlflow.py]: the formatting has to be like I showed with the new line after "Return"

Comment (bamblebam) [reply pytorch_lightning/loggers/mlflow.py]: Yeah I fixed it.

File: pytorch_lightning/loggers/mlflow.py [hunk 3]
```diff
@@ -171,14 +171,24 @@ def experiment(self) -> MlflowClient:
         return self._mlflow_client
 
     @property
-    def run_id(self):
-        # create the experiment if it does not exist to get the run id
+    def run_id(self) -> str:
+        """
+        Create the experiment if it does not exist to get the run id.
+
+        Return:
```
Comment (ethanwharris) [inline pytorch_lightning/loggers/mlflow.py:178]:
```suggestion
        Returns:
```

File: pytorch_lightning/loggers/mlflow.py [hunk 4]
```diff
@@ -171,14 +171,24 @@ def experiment(self) -> MlflowClient:
...
         return self._run_id
 
     @property
-    def experiment_id(self):
-        # create the experiment if it does not exist to get the experiment id
+    def experiment_id(self) -> str:
+        """
+        Create the experiment if it does not exist to get the experiment id.
+
+        Return:
```
Comment (ethanwharris) [inline pytorch_lightning/loggers/mlflow.py:189]:
```suggestion
        Returns:
```

File: pytorch_lightning/loggers/mlflow.py [hunk 5]
```diff
@@ -239,8 +249,20 @@ def save_dir(self) -> Optional[str]:
 
     @property
     def name(self) -> str:
+        """
+        Get the experiment id.
+
+        Return:
```
Comment (ethanwharris) [inline pytorch_lightning/loggers/mlflow.py:255]:
```suggestion
        Returns:
```

File: pytorch_lightning/loggers/mlflow.py [hunk 6]
```diff
@@ -239,8 +249,20 @@ def save_dir(self) -> Optional[str]:
...
+            The experiment id.
+        """
         return self.experiment_id
 
     @property
     def version(self) -> str:
+        """
+        Get the run id.
+
+        Return:
```
Comment (ethanwharris) [inline pytorch_lightning/loggers/mlflow.py:265]:
```suggestion
        Returns:
```

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by ananthsub — APPROVED** [review]

**Pull request merged**